internal/search/job/jobutil/exhaustive_job.go

```
@@ -59,8 +52,49 @@ func NewExhaustive(inputs *search.Inputs) (Exhaustive, error) {
 	}
 
 	repoOptions := toRepoOptions(b, inputs.UserSettings)
-	resultTypes := computeResultTypes(b, inputs.PatternType)
-	planJob := NewTextSearchJob(b, inputs, resultTypes, repoOptions)
+	resultTypes := computeResultTypes(b, inputs.PatternType, result.TypePath|result.TypeFile)
+
+	supportedTypes := result.TypeCommit | result.TypeDiff | result.TypeFile | result.TypePath
+	if resultTypes.Without(supportedTypes) != 0 {
+		return Exhaustive{}, errors.Errorf("your query contains the following type filters: %v. However Search Jobs only supports: %v.", resultTypes, supportedTypes)
+	}
+
+	var planJob job.Job
+
+	if resultTypes.Has(result.TypeCommit | result.TypeDiff) {
+		_, _, own := isOwnershipSearch(b)
+		diff := resultTypes.Has(result.TypeDiff)
+		// Follows the logic of interactive search, see
+		// https://github.com/sourcegraph/sourcegraph/pull/35741
+		repoOptionsCopy := repoOptions
+		repoOptionsCopy.OnlyCloned = true
```

Is a repo not being cloned just a point in time situation? How will a user know why repos were skipped?

---

Yes, that should be a temporary state.

We could go either way here TBH, but it feels ok to mirror what interactive search does for consistency. The user can find out which exact repos-revision pairs were searched by checking the logs (downloadable from the UI). This fits well with the compliance use-case because we state exactly what we searched.

If we set `OnlyCloned = false`, search jobs on large instances might fail more often and the user has to look at the logs to find out why. I would feel better about this option if we had a way of reporting partial success to the user.

---

What happens if onlyCloned is false? Will we use searcher or something like that to pull the info?

---

It seems Search Jobs has a different target use case than interactive search, as it's much more focused on compliance and it'd be really bad if a user missed the fact we skipped over certain repos. So ideally our defaults would reflect that, and we'd loudly surface errors/ missing data instead of focusing on "best effort" as we usually do.

Am I understanding right that we'd currently fail the entire job if one repo search errors? Then your current approach makes sense to me. But big +1 to adding support for partial results so we can clearly surface errors to users.

---

But is that what this setting will do? will onlyCloned just silently filter out non-indexed items? 
And oppositely, if we did onlyCloned=false, would it just work but be slow?

---

Ah sorry, missed your question. My understanding: `onlyCloned=true` will silently skip over non-indexed repos. If `onlyCloned=false`, the search will fail. Searcher is not able to help here, since we don't even have the repo available within Sourcegraph.